### DISCUSSION TITLE
Multiple xauth/eap secrets with the same xauth_id

### DISCUSSION THREAD

**Discussion by martinprejsa** [strongswan/strongswan#2377] [Q&A] [answered]
Is it possible to define multiple xauth/eap secrets with the same id? I have several connections to different sites (in different conf files), but they all use the same id `m.prejsa` in the xauth auth round. It seems that the credentials used are always from the last file loaded.

**Comment by tobiasbrunner** [comment]
No

**Reply by martinprejsa** [reply]
Can i ask why?

**Reply by tobiasbrunner** [reply]
It might actually be possible, try specifying the server identity in addition to the username with your secrets.

**Reply by martinprejsa** [reply]
I've tried creating unique IKE Identities (using the `id` option), and then specifying`xauth_id` in the local xauth round. Hoping that the secret will be matched using the `id` and not the `xauth_id` option.  No luck.

> try specifying the server identity in addition to the username

I don't quite get what you mean by that.

**Reply by tobiasbrunner** [reply]
Read the docs on how to configure secrets.

> I've tried creating unique IKE Identities (using the `id` option), and then specifying`xauth_id` in the local xauth round. Hoping that the secret will be matched using the `id` and not the `xauth_id` option. No luck.

If you use XAuth, the client's IKE identity is pretty much irrelevant later on, the XAuth identity replaces it (same with the EAP identity when using IKEv2).

> I don't quite get what you mean by that.

Just add something like `id-server = <server id>` to your secret (see [the docs](https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html#_secrets_eapsuffix)).

**Reply by martinprejsa** [reply]
> Just add something like id-server = <server id> to your secret (see [the docs](https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html#_secrets_eapsuffix)).

"Multiple unique identities may be specified, each having an `id` prefix if a secret is shared between multiple users"
I'm not sharing a secret between multiple users. I want multiple  different secrets with the same `id` option.

**Reply by tobiasbrunner** [reply]
That's why you add the server identity, not a second client identity.

**Reply by martinprejsa** [reply]
> That's why you add the server identity, not a second client identity.

So if i understand correctly, something like this?
```
connections 
{
    site1 
    {
        local-xauth 
        {
             id = "site1" 
        }
    }

    site2
    {
        local-xauth
        {
             id = "site2"
        }
     }
}

secrets {
   xauth-site1 {
        id = "username"
        id-server = "site1"
        secret = "secret1"
   }
   
   xauth-site2 {
       id = "username"
       id-server = "site2"
       secret = "secret2"
   }
}
```

**Reply by tobiasbrunner** [reply]
No, the server identity is the **remote** IKE identity.

**Reply by martinprejsa** [reply, marked as answer]
> No, the server identity is the **remote** IKE identity.

Yes, that worked. Thanks a lot.

Updated config
```
connections 
{
    site1 
    {
        local-xauth 
        {
             id = "site1" 
        }
        remote
        {
            id = 1.1.1.1
        }
    }

    site2
    {
        local-xauth
        {
             id = "site2"
        }
        remote
        {
             id = 2.2.2.2
         }
     }
}

secrets {
   xauth-site1 {
        id = "username"
        id-server = 1.1.1.1
        secret = "secret1"
   }
   
   xauth-site2 {
       id = "username"
       id-server = 2.2.2.2
       secret = "secret2"
   }
}
```
